### PR TITLE
[type:bug]Sync data error when disable metaData.

### PR DESCRIPTION
**Pull request by KevinClair**
Fix bug, when disable metaData.
```java
public String enabled(final List<String> ids, final Boolean enabled) {
        
        Set<String> idSet = Optional.ofNullable(ids).orElseGet(ArrayList::new)
                .stream().filter(StringUtils::isNotEmpty).collect(Collectors.toSet());
        if (CollectionUtils.isEmpty(idSet)) {
            return AdminConstants.ID_NOT_EXIST;
        }
        List<MetaDataDO> metaDataDoList = Optional.ofNullable(metaDataMapper.selectByIdSet(idSet)).orElseGet(ArrayList::new);
        if (idSet.size() != metaDataDoList.size()) {
            return AdminConstants.ID_NOT_EXIST;
        }
        List<MetaData> metaDataList = metaDataDoList.stream()
                .map(MetaDataTransfer.INSTANCE::mapToData)
                .collect(Collectors.toList());
        metaDataMapper.updateEnableBatch(idSet, enabled);
        
        // the element in metaDataList still be enabled when disable metaData
        eventPublisher.publishEvent(new DataChangedEvent(ConfigGroupEnum.META_DATA, DataEventTypeEnum.UPDATE,
                metaDataList));
        
        return StringUtils.EMPTY;
    }
```

Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by yu199195** [comment]
pls resolve conficits, and send mail to dev@apache.shenyu.org,  

Because

1. I am not sure it is correct to modify

2. it allows other developers to come together for better discussion

**Comment by KevinClair** [comment]
> pls resolve conficits, and send mail to [dev@apache.shenyu.org](mailto:dev@apache.shenyu.org),
> 
> Because
> 
> 1. I am not sure it is correct to modify
> 2. it allows other developers to come together for better discussion

done

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/MetaDataServiceImpl.java [hunk 1]
```diff
@@ -131,41 +130,33 @@ public String update(final MetaDataDTO metaDataDTO) {
...
-        if (idSet.size() != metaDataDoList.size()) {
+        List<MetaDataDO> metaDataDoList = Optional.ofNullable(metaDataMapper.selectByIdList(ids)).orElseGet(ArrayList::new);
+        if (CollectionUtils.isEmpty(metaDataDoList)) {
             return AdminConstants.ID_NOT_EXIST;
         }
         List<MetaData> metaDataList = metaDataDoList.stream()
-                .map(MetaDataTransfer.INSTANCE::mapToData)
-                .collect(Collectors.toList());
-        metaDataMapper.updateEnableBatch(idSet, enabled);
+            .map(metaDataDo -> MetaDataTransfer.INSTANCE.mapToDataEnabled(metaDataDo, enabled))
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/MetaDataServiceImpl.java:154]: .map(metaDataDo -> {
metaDataDo.setEnable(enabled)
 MetaDataTransfer.INSTANCE.mapToData)
}

will better?

Comment (KevinClair) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/MetaDataServiceImpl.java:154]: good idea.

**Review by yu199195 — APPROVED** [review]

**Pull request merged**